src/CTA.FeatureDetection.ProjectType/CompiledFeatures/VBClassLibraryFeature.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Codelyzer.Analysis;
+using CTA.FeatureDetection.Common.Extensions;
+using CTA.FeatureDetection.Common.Models.Features.Base;
+using CTA.Rules.Config;
+using Microsoft.Build.Construction;
+
+namespace CTA.FeatureDetection.ProjectType.CompiledFeatures
+{
+    public class VBClassLibraryFeature : CompiledFeature
+    {
+        /// <summary>
+        /// Determines if a project is an VB class library project by looking at file extensions and nuget references
+        /// </summary>
+        /// <param name="analyzerResult"></param>
+        /// <returns>Whether a project is an VB class library project or not</returns>
+        public override bool IsPresent(AnalyzerResult analyzerResult)
+        {
+            var project = analyzerResult.ProjectResult;
+            var isVBProj = project.ContainsFileWithExtension(Constants.VbClassExtension, true) && project.ProjectFilePath.EndsWith(Constants.VbProjExtension)
+                            &&!(project.ContainsDependency(Constants.MvcReferenceIdentifier) && project.DeclaresClassBlocksWithBaseType(Constants.MvcControllerOriginalDefinition))
+                            &&!((project.ContainsNugetDependency(Constants.WebApiNugetReferenceIdentifier) || project.ContainsDependency(Constants.WebApiReferenceIdentifier))
+                            && project.DeclaresClassBlocksWithBaseType(Constants.WebApiControllerOriginalDefinition));
```

Comment:
Just double checking we want true for WebApiControllerOriginalDefinition in the ClassLibraryFeature class.

Reply:
! condition for the api check, since api, webforms and mvc are also library projects as output type